core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1437,17 +1437,20 @@ class KafkaApis(val requestChannel: RequestChannel,
     def sendResponseCallback(producerId: Long, result: TransactionResult)(responseStatus: Map[TopicPartition, PartitionResponse]): Unit = {
       errors.put(producerId, responseStatus.mapValues(_.error).asJava)
 
-      val successfulPartitions = responseStatus.filter { case (_, partitionResponse) =>
-        partitionResponse.error == Errors.NONE
-      }.keys.toSeq
+      val offsetsPartitions = responseStatus.filterKeys(_.topic == GROUP_METADATA_TOPIC_NAME)
+      if (offsetsPartitions.nonEmpty) {
+        val successfulOffsetsPartitions = offsetsPartitions.filter { case (_, partitionResponse) =>
```

Comment:
What if there are no `successfulOffsetsPartitions`? Would it make sense to do the filter after the `filterKeys` and only enter this `if` if the result of that is non empty?